### PR TITLE
ROLLUP-102/Fix CI on GitHub

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Fix Ci on Github Actions
- Merge the integration and unit testing
- Simplify lint action
- Run the integration and unit testing just when PR is ready

## Why

- We need to validate the changes to avoid changing the current behavior, unless that is necessary

## Refs

- [ROLLUP-102](https://rsklabs.atlassian.net/browse/ROLLUP-102)

### CODE REVIEW

File: .github/workflows/ci-iov-integration.yml [hunk 1]
```diff
@@ -1,21 +1,14 @@
 name: ri-aggregation CI Integration
-on:
-  push:
-    branches:
-    - rsk_merge_master_Dec2021
-    #- rsk_port_initial
+on: [push]
```
Comment (franciscotobar) [inline .github/workflows/ci-iov-integration.yml:2]: test

Comment (franciscotobar) [inline .github/workflows/ci-iov-integration.yml:2]:
```suggestion
on: [push]
```

File: .github/workflows/ci-iov-integrations.yml [hunk 2]
```diff
@@ -1,21 +1,18 @@
-name: ri-aggregation CI Integration
-on:
-  push:
-    branches:
-    - rsk_merge_master_Dec2021
-    #- rsk_port_initial
+name: rif-rollup CI integration tests
```
Comment (AndresQuijano) [inline .github/workflows/ci-iov-integrations.yml:1]: Random comment

File: .github/actions/draft-pr/action.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: Mark PR as Draft
+description: Converts the given pull request to a draft
```
Comment (franciscotobar) [inline .github/actions/draft-pr/action.yml:2]: random comment

File: .github/actions/draft-pr/action.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (franciscotobar) [inline .github/actions/draft-pr/action.yml:3]: random

File: .github/actions/draft-pr/action.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+inputs:
+  token:
```
Comment (franciscotobar) [inline .github/actions/draft-pr/action.yml:5]: random

File: .github/actions/draft-pr/action.yml (hunk 4 above)
Comment (franciscotobar) [inline .github/actions/draft-pr/action.yml:3]: test

File: .github/actions/draft-pr/action.yml (hunk 3 above)
Comment (franciscotobar) [inline .github/actions/draft-pr/action.yml:2]: test

File: .github/actions/draft-pr/action.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+name: Mark PR as Draft
+description: Converts the given pull request to a draft
+
+inputs:
```
Comment (franciscotobar) [inline .github/actions/draft-pr/action.yml:4]: test

File: .github/actions/draft-pr/action.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  token:
+    required: true
+    description: Github Token used to make API request to mutate pull requests
+    default: ${{ github.token }}
+  pull-request-node-id:
```
Comment (AndresQuijano) [inline .github/actions/draft-pr/action.yml:9]:
```suggestion
  pull-request-node-id: fake comment
```

File: .github/actions/draft-pr/action.yml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Mark as draft
+      shell: bash
+      run: |
+        echo ${{ inputs.token }} | gh auth login --with-token
+        gh api graphql -F id=${{ inputs.pull-request-node-id }}  -f query='
+          mutation($id: ID!) {
+            convertPullRequestToDraft(input: { pullRequestId: $id }) {
+              pullRequest {
+                id
+                number
```
Comment (AndresQuijano) [inline .github/actions/draft-pr/action.yml:26]:
```suggestion
                number fake comment
```

**Review by AndresQuijano — COMMENTED** [review]
sfsdf

**Review by antomor — CHANGES_REQUESTED** [review]
Request changes comment

File: docker-compose-runner.yml [hunk 9]
```diff
@@ -28,8 +28,7 @@ services:
       - ./etc/tokens/:/etc/tokens
 
   zk:
-    build:
-      context: ./docker/zk-environment 
+    image: "franciscotobar/rollup-zk:0.0.1"
```
Comment (franciscotobar) [inline docker-compose-runner.yml:31]: Waiting for image to be published in the RSK docker hub

Comment (franciscotobar) [reply docker-compose-runner.yml]: updated

**Review by antomor — CHANGES_REQUESTED** [review]
Testing "request changes"

File: .github/archives/ci.yml
Comment (antomor) [inline .github/archives/ci.yml:1]: Since we're not using it, do we want to delete this file?

File: .github/archives/deploy.rinkeby-beta.yml
Comment (antomor) [inline .github/archives/deploy.rinkeby-beta.yml:1]: Since we're not using it, do we want to delete this file?

File: .github/archives/loadtest.yml
Comment (antomor) [inline .github/archives/loadtest.yml:1]: Since we're not using it, do we want to delete this file?

File: .github/workflows/ci-iov.yml [hunk 10]
```diff
@@ -57,4 +68,46 @@ jobs:
...
+          ci_run zk test i api-docs
+
+      - name: rust-sdk-integration-tests
+        run: ci_run zk test i rust-sdk
+
+      - name: withdrawal-helpers-integration-tests
+        run: ci_run zk test i withdrawal-helpers
+
+      - name: Show logs
+        if: always()
```
Comment (antomor) [inline .github/workflows/ci-iov.yml:109]: Do we need this `if` here?

File: core/bin/zksync_api/src/fee_ticker/validator/mod.rs [hunk 11]
```diff
@@ -190,40 +190,40 @@ mod tests {
 
     #[tokio::test]
     async fn check_tokens() {
-        let dai_token_address =
-            Address::from_str("6b175474e89094c44da98b954eedeac495271d0f").unwrap();
-        let dai_token = Token::new(TokenId(1), dai_token_address, "DAI", 18, TokenKind::ERC20);
-        let phnx_token_address =
-            Address::from_str("38A2fDc11f526Ddd5a607C1F251C065f40fBF2f7").unwrap();
-        let phnx_token = Token::new(TokenId(2), phnx_token_address, "PHNX", 18, TokenKind::ERC20);
+        let rif_token_address =
```
Comment (antomor) [inline core/bin/zksync_api/src/fee_ticker/validator/mod.rs:193]: Where does this address come from?

File: core/bin/zksync_api/src/fee_ticker/validator/mod.rs [hunk 12]
```diff
@@ -190,40 +190,40 @@ mod tests {
...
+            Address::from_str("0x2acc95758f8b5f583470ba265eb685a8f45fc9d5").unwrap();
+        let rif_token = Token::new(TokenId(1), rif_token_address, "RIF", 18, TokenKind::ERC20);
+        let rdoc_token_address =
+            Address::from_str("0x2d919f19D4892381d58EdEbEcA66D5642ceF1A1F").unwrap();
```
Comment (antomor) [inline core/bin/zksync_api/src/fee_ticker/validator/mod.rs:197]: Where does this address come from?

File: core/bin/zksync_event_listener/src/monitor.rs [hunk 13]
```diff
@@ -52,6 +52,8 @@ impl Handler<RemoveSubscriber> for ServerMonitor {
 impl Handler<NewEvents> for ServerMonitor {
     type Result = ();
 
+    //FIXME remove clippy rule.
```
Comment (antomor) [inline core/bin/zksync_event_listener/src/monitor.rs:55]: @franciscotobar Do you think we should create a task to address it?

Comment (antomor) [reply core/bin/zksync_event_listener/src/monitor.rs:55]:
```suggestion
    // FIXME: remove clippy rule.
```

File: docker/zk-environment/Dockerfile [hunk 14]
```diff
@@ -11,23 +11,24 @@ RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debia
...
 # Install Rust and required cargo packages
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH
 RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
-RUN cargo install diesel_cli --no-default-features --features postgres
+RUN rustup install 1.69.0
+RUN cargo install diesel_cli  --version=1.4.0 --no-default-features --features postgres
 RUN cargo install --version=0.5.6 sqlx-cli
+RUN cargo install wasm-pack --git https://github.com/d3lm/wasm-pack --rev 713868b204f151acd1989c3f29ff9d3bc944c306
```
Comment (antomor) [inline docker/zk-environment/Dockerfile:31]: Do we need this particular version of `wasm-pack`?

File: docker/zk-environment/Dockerfile [hunk 15]
```diff
@@ -40,4 +41,4 @@ ENV ZKSYNC_HOME=/usr/src/zksync
 ENV PATH="${ZKSYNC_HOME}/bin:${PATH}"
 ENV CI=0
 RUN cargo install sccache
-ENV RUSTC_WRAPPER=/usr/local/cargo/bin/sccache
+ENV RUSTC_WRAPPER=/usr/local/cargo/bin/sccache
```
Comment (antomor) [inline docker/zk-environment/Dockerfile:44]: Added an empty line at the end
```suggestion
ENV RUSTC_WRAPPER=/usr/local/cargo/bin/sccache

```

**Review by antomor — COMMENTED** [review]
I haven't tried it locally yet, but apart from the comments I left, LGTM

File: .github/workflows/ci-iov.yml (hunk 10 above)
Comment (franciscotobar) [reply .github/workflows/ci-iov.yml:109]: yes we do, you can read the definition here. https://docs.github.com/en/actions/learn-github-actions/expressions#always.

File: .github/archives/loadtest.yml
Comment (franciscotobar) [reply .github/archives/loadtest.yml:1]: I could delete more of them, if you want we can have a meeting and I will delete all the one that I think that are unnecessary.

File: core/bin/zksync_api/src/fee_ticker/validator/mod.rs (hunk 11 above)
Comment (franciscotobar) [reply core/bin/zksync_api/src/fee_ticker/validator/mod.rs:193]: is the address in `etc>tokens>mainnet.json`

File: core/bin/zksync_api/src/fee_ticker/validator/mod.rs (hunk 12 above)
Comment (franciscotobar) [reply core/bin/zksync_api/src/fee_ticker/validator/mod.rs:197]: is the address in `etc>tokens>mainnet.json`

File: core/bin/zksync_event_listener/src/monitor.rs (hunk 13 above)
Comment (franciscotobar) [reply core/bin/zksync_event_listener/src/monitor.rs:55]: We could, I dont think that is possible to apply the rule but a task could be created to prove that Im wrong :S

File: docker/zk-environment/Dockerfile (hunk 14 above)
Comment (franciscotobar) [reply docker/zk-environment/Dockerfile:31]: Ill be removing this file from the PR.

File: docker/zk-environment/Dockerfile (hunk 15 above)
Comment (franciscotobar) [reply docker/zk-environment/Dockerfile:44]: Ill be removing this file from the PR.

File: core/bin/zksync_api/src/fee_ticker/validator/mod.rs (hunk 12 above)
Comment (franciscotobar) [reply core/bin/zksync_api/src/fee_ticker/validator/mod.rs]: update

File: docker/zk-environment/Dockerfile (hunk 14 above)
Comment (franciscotobar) [reply docker/zk-environment/Dockerfile]: update

File: core/bin/zksync_event_listener/src/monitor.rs (hunk 13 above)
Comment (antomor) [reply core/bin/zksync_event_listener/src/monitor.rs]: Create task [Rollup-214](https://rsklabs.atlassian.net/browse/ROLLUP-214)

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**